### ISSUE TITLE
tools.scholiaanalytics trying to run custom Docker images on the Toolforge Kubernetes cluster

### ISSUE THREAD

**Issue by supertassu** [WDscholia/scholia#1432]
Hi,

The `scholiaanalytics` Toolforge tool is trying (and failing) to run containers with images from docker hub (`nunodoc/scholia-analytics` and `nginx`), however as described in https://wikitech.wikimedia.org/wiki/Help:Toolforge/Kubernetes#Container_images currently only images in the [internal docker registry](https://docker-registry.toolforge.org/) can be used.

I didn't delete those Kubernetes deployment objects since I did not find the files used to create them anywhere and wanted to make sure you have everything important saved. Could you please make sure that nothing important would be lost and then either fix them to use Toolforge images or delete those objects if not possible? Is there anything that you would need but is not possible with the Toolforge images?

**Comment by fnielsen** [comment]
I am under the impression that Toolforge is using their own container system and one cannot start ones own container. But I might have misunderstood something.

**Comment by supertassu** [comment]
> one cannot start ones own container

Yeah, this is generally the case. However Kubernetes is a fairly complex thing, and while we block the creation of new containers (Pods in Kubernetes language) with non-approved images, it does not prevent creation of Deployment objects with custom containers. A Deployment is an abstraction around ReplicaSet objects, which are an abstraction that make sure there are configured Pods running.

So while we block the creation of the actual containers (pods), there are hand-made deployment objects in the "scholiaanalytics" tool that are constantly trying and failing to create pods with non-approved images. I found those when performing routine Toolforge maintenance and created this issue to make sure you (maintainers of that tool) know about this and can potentially do something about that, for example delete those deployment objects or change them to use approved images.

Is this explanation clear enough?

**Comment by Daniel-Mietchen** [comment]
Perhaps @nunogit can comment?
